### ISSUE TITLE
Cannot find package '@nuxt/ui-templates'

### ISSUE THREAD

**Issue by heyitsfranklin** [nuxt/framework#3608]
### Environment

------------------------------
- Operating System: `Windows_NT`
- Node Version:     `v14.18.1`
- Nuxt Version:     `2.16.0-27358576.777a4b7f`
- Package Manager:  `npm@6.14.15`
- Bundler:          `Webpack`
- User Config:      `server`, `head`, `css`, `plugins`, `serverMiddleware`, `components`, `buildModules`, `axios`, `modules`, `bootstrapVue`, `build`, `auth`, `router`, `bridge`
- Runtime Modules:  `@nuxtjs/axios@5.13.6`, `@nuxtjs/auth-next@5.0.0-1641544397.1af946a`, `bootstrap-vue/nuxt`, `cookie-universal-nuxt@2.1.5`, `@nuxtjs/moment@1.6.1`
- Build Modules:    `@nuxtjs/eslint-module@2.0.0`, `@nuxt/bridge@3.0.0-27449998.32cf973`
------------------------------

### Reproduction

1. Upgraded to the latest version of nuxt-edge and nuxt-bridge
2. `npm install`
3. Ran `nuxi dev`
4. Error is thrown

### Describe the bug

Issue occurs when attempting to run `nuxi dev` after upgrading to the latest version of nuxt-edge and nuxt-bridge

### Additional context

_No response_

### Logs

```shell
Nuxt CLI v3.0.0-27449998.32cf973                                                                                                                                                                                13:20:20
                                                                                                                                                                                                                13:20:20
  > Local:    http://localhost:3000/
  > Network:  http://169.254.139.185:3000/
  > Network:  http://192.168.56.1:3000/
  > Network:  http://192.168.0.129:3000/


 FATAL  Cannot find package '@nuxt/ui-templates' imported from C:\repo\frontend\node_modules\@nuxt\nitro\dist\index.mjs                                                             13:20:23

  at new NodeError (internal/errors.js:322:7)
  at packageResolve (internal/modules/esm/resolve.js:687:9)
  at moduleResolve (internal/modules/esm/resolve.js:728:18)
  at Loader.defaultResolve [as _resolve] (internal/modules/esm/resolve.js:842:11)
  at Loader.resolve (internal/modules/esm/loader.js:89:40)
  at Loader.getModuleJob (internal/modules/esm/loader.js:242:28)
  at ModuleWrap.<anonymous> (internal/modules/esm/module_job.js:76:40)
  at link (internal/modules/esm/module_job.js:75:36)


 ERROR  Cannot start nuxt:  Cannot find package '@nuxt/ui-templates' imported from C:\repo\frontend\node_modules\@nuxt\nitro\dist\index.mjs                                         13:20:23

  at new NodeError (internal/errors.js:322:7)
  at packageResolve (internal/modules/esm/resolve.js:687:9)
  at moduleResolve (internal/modules/esm/resolve.js:728:18)
  at Loader.defaultResolve [as _resolve] (internal/modules/esm/resolve.js:842:11)
  at Loader.resolve (internal/modules/esm/loader.js:89:40)
  at Loader.getModuleJob (internal/modules/esm/loader.js:242:28)
  at ModuleWrap.<anonymous> (internal/modules/esm/module_job.js:76:40)
  at link (internal/modules/esm/module_job.js:75:36)


 ERROR  Cannot read property 'callHook' of undefined                                                                                                                                                            13:20:23  

  at Object.invoke (/C:/repo/frontend/node_modules/nuxi/dist/chunks/dev.mjs:6773:31)
  at async _main (/C:/repo/frontend/node_modules/nuxi/dist/chunks/index.mjs:384:7)
```

**Comment by heyitsfranklin** [comment]
Update...

Running `npm install -d @nuxt/ui-templates@npm:@nuxt/ui-templates-edge@latest` solves the bug.

Can someone please fix this dependency?

**Comment by danielroe** [comment]
I can't replicate this. As you can see, it's in the nitro package.json:

https://github.com/nuxt/framework/blob/7650f6c57dd6b401e62b944731596642ec5f953f/packages/nitro/package.json#L15-L20

I would guess this would be solved by refreshing your package lockfile.

Can you provide a reproduction?

**Comment by heyitsfranklin** [comment]
> I can't replicate this. As you can see, it's in the nitro package.json:
> 
> https://github.com/nuxt/framework/blob/7650f6c57dd6b401e62b944731596642ec5f953f/packages/nitro/package.json#L15-L20
> 
> I would guess this would be solved by refreshing your package lockfile.
> 
> Can you provide a reproduction?

Looks like the issue is local to my environment then. 

I tried deleting package-lock.json, deleting the node_modules directory, and then running npm install (without @nuxt/ui-templates in my package.json) and the package never gets installed. I'll try to debug why this is happening.

**Comment by danielroe** [comment]
Might be a package manager issue (worth trying with a more recent version of npm).
I also assume you have `nuxi` installed locally in your project, but if not you could try doing that.

**Comment by heyitsfranklin** [comment]
> Might be a package manager issue (worth trying with a more recent version of npm). I also assume you have `nuxi` installed locally in your project, but if not you could try doing that.

I tried that and it didn't seem to have any effect.

I tried a variety of things (updating nvm, upgrading npm but failing the install) and somehow everything is working again.

Cheers for the help!

**Comment by ataylor32** [comment]
I'm also running into this and I'm also using Node 14. This is what I tried:

```bash
npx nuxi init nuxt3-app && cd nuxt3-app && npm install && npm run dev
```

And this is the error I got:

```
Failed to parse source for import analysis because the content contains invalid JS syntax. Install @vitejs/plugin-vue to handle .vue files.
```

After running `npm install @vitejs/plugin-vue`, I get a new error:

```
Cannot start nuxt:  Cannot find package '@nuxt/ui-templates' imported from /path/to/nuxt3-app/node_modules/@nuxt/nitro/dist/index.mjs
```

https://v3.nuxtjs.org/getting-started/introduction indicates that Node 14 is supported. If that's not actually the case then it should be updated at some point.

**Comment by danielroe** [comment]
I'm using Node 14 too, with no issues. Try using npm@8 or yarn. Does that resolve this for you?

**Comment by ataylor32** [comment]
@danielroe I upgraded npm (I'm now using 8.5.4) and that fixed the problem. Thank you!

**Comment by danielroe** [comment]
This seems to be a general issue with older versions of `npm`.

The issue here is your version of npm is not installing `@nuxt/ui-templates` as a dependency of `nuxt3` or `@nuxt/bridge`.

You can verify this by running `npm list @nuxt/ui-templates`. Here's sample output.

```bash
❯ npm list @nuxt/ui-templates
/tmp/nuxt3-app
└─┬ nuxt3@3.0.0-27460489.53fbca7
  ├─┬ @nuxt/nitro@npm:@nuxt/nitro-edge@3.0.0-27460489.53fbca7
  │ └── @nuxt/ui-templates@npm:@nuxt/ui-templates-edge@0.0.0-27455683.83d2457  (@nuxt/ui-templates) invalid
  └── UNMET DEPENDENCY @nuxt/ui-templates@npm:@nuxt/ui-templates-edge@latest

npm ERR! missing: @nuxt/ui-templates@npm:@nuxt/ui-templates-edge@latest, required by nuxt3@3.0.0-27460489.53fbca7
npm ERR! invalid: @nuxt/ui-templates-edge@0.0.0-27455683.83d2457 /tmp/nuxt3-app/node_modules/@nuxt/nitro/node_modules/@nuxt/ui-templates
```

Note that it is in `nuxt3`'s dependencies: https://github.com/nuxt/framework/blob/fb150825ceebbffddd49b312962bef9e6f3ed1b5/packages/nuxt3/package.json#L32-L36

I'm not sure what we can do from a nuxt perspective, but you can either use `npm@8` or manually run `npm install @nuxt/ui-templates@npm:@nuxt/ui-templates-edge@latest`.

**Comment by cam424** [comment]
> Update...
> 
> Running `npm install -d @nuxt/ui-templates@npm:@nuxt/ui-templates-edge@latest` solves the bug.
> 
> Can someone please fix this dependency issue?

This fixed it for me. 
For context, I just did a fresh install with `npx nuxi init nuxt3-app`, running node 16.14.2, after install an error said that I needed to install `@vitejs/plugin-vue`, then ran into another error saying I was now missing `@nuxt/ui-templates` as a dependency for `@nuxt/nitro`.

Running the above command resolved the issue with the dependency and now I'm up and running without error.
